From 40dabfcbcea56f204a216368914ae346b96d0531 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 26 Nov 2015 15:28:36 +0800
Subject: [PATCH 0696/5242] MLK-11911-1 mxc IPUv3: capture: Define div_ratio
 as type of int32_t

commit  b971b58f488289c7ff754574c19e1aee7af9e987 from
https://source.codeaurora.org/external/imx/linux-imx.git

The local variable div_ratio could be less than zero, so let's define
it as type of int32_t instead of uint32_t.

This issue is reported by Coverity:
Unsigned compared against 0 (NO_EFFECT)
unsigned_compare: This less-than-zero comparison of an unsigned value
is never true. div_ratio < 0U.
	if (div_ratio > 0xFF || div_ratio < 0) {
		dev_dbg(ipu->dev, "value of pixel_clk extends normal range\n");
		return -EINVAL;
	}

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/ipu3/ipu_capture.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/ipu3/ipu_capture.c b/drivers/mxc/ipu3/ipu_capture.c
index 09bb6d9..f304c81 100644
--- a/drivers/mxc/ipu3/ipu_capture.c
+++ b/drivers/mxc/ipu3/ipu_capture.c
@@ -43,7 +43,7 @@
 int _ipu_csi_mclk_set(struct ipu_soc *ipu, uint32_t pixel_clk, uint32_t csi)
 {
 	uint32_t temp;
-	uint32_t div_ratio;
+	int32_t div_ratio;
 
 	div_ratio = (clk_get_rate(ipu->ipu_clk) / pixel_clk) - 1;
 
-- 
1.7.9.5

